lib/Transforms/Scalar/DxilRemoveDeadBlocks.cpp

```diff
@@ -304,6 +330,18 @@ static bool DeleteNonContributingValues(Function &F, DxilValueCache *DVC) {
   return Changed;
 }
 
+static void EnsureDxilModule(Module *M) {
+  if (M->HasDxilModule())
+    return;
+  for (Function &F : *M) {
+    if (OP::IsDxilOpFunc(&F)) {
```

Comment:
Wait, so this only happens if there are dxil intrinsic calls?  This seems a bit subtle.  Is that intentional?  Is this to avoid doing so on HL module or something?

Reply:
> Is this to avoid doing so on HL module or something?
That's right. The thought is maybe this pass should be usable at any stage. The real reason dxil module is needed is because instruction simplify asserts when it hits a dx op but there's no dxil module. Maybe that's what needs to be fixed instead.